### ISSUE TITLE
Breaks Session Manager completely

### ISSUE THREAD

**Issue by afalout** [kashiif/copy-urls-expert#41]
Hi -installed this plugin and found that Session Manager no longer saves sessions correctly. Automatic session backups result in saved sessions with many thousands of (non-existing) tabs which cannot be restored. Even the sessions saved before installing this plugin cannot be loaded any more - resulting error is: 

This operation failed due to a file access error:
window.tabs is undefined(TypeError)

half way restored...-2017-03-13.session

SessionCookiesInternal.getHostsForWindow@resource://app/modules/sessionstore/SessionCookies.jsm:105:1
SessionCookiesInternal.update@resource://app/modules/sessionstore/SessionCookies.jsm:65:19
this.SessionCookies<.update@resource://app/modules/sessionstore/SessionCookies.jsm:28:5
SessionStoreInternal.getCurrentState@resource:///modules/sessionstore/SessionStore.jsm:2762:5
ssi_getBrowserState@resource:///modules/sessionstore/SessionStore.jsm:2001:17
ss_getBrowserState@resource:///modules/sessionstore/SessionStore.jsm:217:12
Private.getSessionState@chrome://sessionmanager/content/modules/session_data_processing.jsm:61:81
this.SessionDataProcessing.getSessionState@chrome://sessionmanager/content/modules/session_data_processing.jsm:32:10
... etc etc ...

Disabling this plugin did not seem to help - had to remove it completely.

Really loved the plugin, but unfortunately have to choose, and session backup is of higher priority for me.

Thanks,
Andrej

**Comment by kashiif** [comment]
I would recommend creating a new profile and installing both Copy Urls
Expert and Session Manager in that profile. Then restore the session. Make
sure that there is no other extension installed in that new profile.

Copy Urls Expert does not modify Firefox default behavior in any way.  I
have received complaints in the past of issues similar to this but every
time there is some another extension causing incompatability. I can confirm
that I use both CUE and Session Manager together and never had problems in
restoring sessions.

**Comment by afalout** [comment]
Hi Kashif, thanks for your reply; Having CUE work in a separate profile without all the plugins I need wont help me - even if it worked. Here are some details because of which I believe that CUE is in some way involved in this:

1 - I have Firefox installed on 11 machines, including 2 Linux machines , 2 Android tablets and Windows
2 - All have sync activated, so exactly the same plugins are installed, with exactly the same settings propagated across all of them
3 - I did not change any Firefox settings, plugins, or plugins settings for at least 5 months. All machines and all Firefox installations in that time did not have absolutely any issues.
4 - After installing CUE on >ONE< Linux machine, that instance of Firefox started crashing every few hours (that is how I discovered that Session Manager "Restore Session" does not work any more). As none of the other Firefox instances where restarted in that time, CUE was downloaded by sync - but not installed/activated - so they continued to work fine.
5 - When I removed CUE from that one Linux machine, after 3 days of crashing and SM issues, Firefox crashing stopped immediately, and Session Manager started working again.
6 - Since that time 4 days passed without a single Firefox crash.

Now I do appreciate that you believe that CUE does not modify Firefox behaviour in any way, but I would say that something, somewhere, somehow related to CUE does - directly of indirectly - does.

I often have large number of windows open in FF (10-15) and on average 200 tabs open at any time. If that matters. I noticed that sessions saved that would not load show between 3000 and 5000 tabs on them. Trying to open ANY saved sessions while CUE was installed would restore all windows, but only first window will have tabs restored. 

Not sure this is of any help to you, but I hope you will keep it in mind and possibly one day spot something that might cause described behaviour.

All the best,
Andrej
